packages/scripts/src/monitor-oa/common.ts

```diff
@@ -0,0 +1,111 @@
+import { getRetryProvider } from "@uma/common";
+import { ERC20Ethers } from "@uma/contracts-node";
+import { delay } from "@uma/financial-templates-lib";
+import { utils } from "ethers";
+import { getContractInstanceWithProvider } from "../utils/contracts";
+
+import type { Provider } from "@ethersproject/abstract-provider";
+
+export interface BotModes {
+  assertionsEnabled: boolean;
+  disputesEnabled: boolean;
+  settlementsEnabled: boolean;
+}
+
+export interface BlockRange {
+  start: number;
+  end: number;
+}
+
+export interface MonitoringParams {
+  provider: Provider;
+  chainId: number;
+  blockRange: BlockRange;
+  pollingDelay: number;
+  botModes: BotModes;
+}
+
+export const initMonitoringParams = async (env: NodeJS.ProcessEnv): Promise<MonitoringParams> => {
+  if (!env.CHAIN_ID) throw new Error("CHAIN_ID must be defined in env");
+  const chainId = Number(env.CHAIN_ID);
+
+  if (!chainId || (chainId != 1 && chainId != 5)) throw new Error("This script should be run on mainnet or goerli");
+
+  // Creating provider will check for other chainId specific env variables.
+  const provider = getRetryProvider(chainId) as Provider;
+
+  // Default to 1 minute polling delay.
+  const pollingDelay = env.POLLING_DELAY ? Number(env.POLLING_DELAY) : 60;
+
+  if (pollingDelay === 0 && (!env.STARTING_BLOCK_NUMBER || !env.ENDING_BLOCK_NUMBER)) {
+    throw new Error("Must provide STARTING_BLOCK_NUMBER and ENDING_BLOCK_NUMBER if running serverless");
+  }
+
+  // If no block numbers are provided, default to the latest block.
+  const latestBlockNumber: number = await provider.getBlockNumber();
+  const startingBlock = env.STARTING_BLOCK_NUMBER ? Number(env.STARTING_BLOCK_NUMBER) : latestBlockNumber;
+  const endingBlock = env.ENDING_BLOCK_NUMBER ? Number(env.ENDING_BLOCK_NUMBER) : latestBlockNumber;
+  if (startingBlock > endingBlock) {
+    throw new Error("STARTING_BLOCK_NUMBER must be less than or equal to ENDING_BLOCK_NUMBER");
+  }
+
+  const botModes = {
+    assertionsEnabled: env.ASSERTIONS_ENABLED === "true",
+    disputesEnabled: env.DISPUTES_ENABLED === "true",
+    settlementsEnabled: env.SETTLEMENTS_ENABLED === "true",
+  };
+
+  return {
+    provider,
+    chainId,
+    blockRange: { start: startingBlock, end: endingBlock },
+    pollingDelay,
+    botModes,
+  };
+};
+
+export const waitNextBlockRange = async (params: MonitoringParams): Promise<BlockRange> => {
```

Comment:
as we develop this new way of building monitor scripts it might be reasonable to move this to some other kind of directory structure where we can re-use this with the `monitor-dvm` package and logic. not something to do now but if we build a 3rd monitor bot using this new technique we should do this.